File: 1_Counting_DNA_Nucleotides/CountingDNANucleotides.cpp
/*
Counting DNA Nucleotides
url: http://rosalind.info/problems/dna/

Given: A DNA string  of length at most 1000 nt
Return: Four integers (separated by spaces) counting the respective number of times that the symbols 'A', 'C', 'G', and 'T' occur in
*/
#include <iostream>
#include <fstream>
#include <istream>
#include <string>

int main(){
    std::cout << "Ciao" << std::endl;
    std::string text;
    std::string dna = "";
    std::ifstream data("../1_Counting_DNA_Nucleotides/rosalind_dna.txt");

    while(getline(data, text)){
        dna = dna + text;
    }

    int a = 0, c = 0, g = 0, t = 0;
    for (int i = 0; i < dna.length(); i++){
        if (dna[i] == 'A'){
            a++;
        }else if (dna[i] == 'C'){
            c++;
        }else if (dna[i] == 'G'){
            g++;
        }else if (dna[i] == 'T'){
            t++;
        }
    }

    std::cout << a << " " << c << " " << g << " " << t << std::endl;     

    data.close();

}

File: DNA_Counting_DNA_Nucleotides/CountingDNANucleotides.py
"""
Counting DNA Nucleotides
url: http://rosalind.info/problems/dna/

Given: A DNA string  of length at most 1000 nt
Return: Four integers (separated by spaces) counting the respective number of times that the symbols 'A', 'C', 'G', and 'T' occur in
"""
import os

dataset = open(os.getcwd()+"/DNA_Counting_DNA_Nucleotides/rosalind_dna.txt", "r")
dna = dataset.readline().strip()

number_a = dna.count("A")
number_c = dna.count("C")
number_g = dna.count("G")
number_t = dna.count("T")

print(f"{number_a} {number_c} {number_g} {number_t}")